target_snowflake/file_formats/parquet.py

```diff
@@ -0,0 +1,122 @@
+"""Parquet file format functions"""
+import os
+import pandas
+
+from typing import Dict, List
+from tempfile import mkstemp
+
+import target_snowflake.flattening as flattening
+
+
+def create_copy_sql(table_name: str,
+                    stage_name: str,
+                    s3_key: str,
+                    file_format_name: str,
+                    columns: List):
+    """Generate a Parquet compatible snowflake COPY INTO command"""
+    return "COPY INTO {} ({}) " \
+           "FROM (SELECT {} FROM '@{}/{}') " \
+           "FILE_FORMAT = (format_name='{}')".format(
+        table_name,
+        ', '.join([c['name'] for c in columns]),
+        ', '.join(["{}($1:{}) {}".format(c['trans'],
+                                         c['json_element_name'],
```

Comment:
Can you elaborate a bit on why is this `json_element_name` needed and used for?

Reply:
Selecting from parquet files requires the column names to be case-sensitive. The currently available `name` key in the dict is an sql friendly quoted and uppercased version of the column, which is fine for CSV but not compatible with selecting data from Parquet files:

**To Select from Parquet:**
```
SELECT ($1:col_1) COL_1, ($1:colTwo) COL_2, parse_json($1:col_3) COL_3 
  FROM '@foo_stage/foo_s3_key.csv' 
  (FILE_FORMAT => 'parquet_file_format')
  ;
```
Parquet files are a one column JSON representation of each record where element names are case-sensitive. We need to select the columns by `$1:caseSensitiveColumnName`. More info [here](https://docs.snowflake.com/en/user-guide/querying-semistructured.html#traversing-semi-structured-data). The case sensive version of the column name is the new `json_element_name` key in the dict.

**To Select from CSV:**
```
SELECT ($1) COL_1, ($2) COL_2, parse_json($3) COL_3
  FROM '@foo_stage/foo_s3_key.csv' 
  (FILE_FORMAT => 'csv_file_format)
```
CSV files are tabular and we can select columns directly by referring columns sequnce, by $1, $2, $n .

Reply:
Very nice explanation, thanks! It's all clear.